TestProjects/Navigation/RoutingExample.Core/ViewModels/TestCViewModel.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Threading.Tasks;
+using MvvmCross.Core.Navigation;
+using MvvmCross.Core.ViewModels;
+using MvvmCross.Platform;
+using RoutingExample.Core.ViewModels;
+
+namespace RoutingExample.Core.ViewModels
+{
+    public class TestCViewModel
+        : MvxViewModel<User, User>
+    {
+        private readonly IMvxNavigationService _navigationService;
+        public TestCViewModel(IMvxNavigationService navigationService)
+        {
+            _navigationService = navigationService;
+        }
+
+        private string _parameter;
+        public string Parameter
+        {
+            get { return _parameter; }
+            set { SetProperty(ref _parameter, value); }
+        }
+
+        public void Init(string viewmodelcparameter)
```

Comment:
Prepare is implemented because the interface is there. It would be great to have a way to make this typed as well. So maybe and a new interface which will receive the parameters?